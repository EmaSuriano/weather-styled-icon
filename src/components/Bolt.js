import styled, { keyframes } from 'styled-components';
import { ThemePropType, defaultTheme } from '../constants';

const BoltAnimation = ({ theme }) => keyframes`
45% {
  color: #fff;
  background: #fff;
  opacity: 0.2;
}
50% {
  color: ${theme.boltColor};
  background: ${theme.boltColor};
  opacity: 1;
}
55% {
  color: #fff;
  background: #fff;
  opacity: 0.2;
}
`;

const Bolt = styled.div.attrs({
  'data-e2e': 'bolt',
})`
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -0.25em 0 0 -0.125em;
  color: grey;
  opacity: 0.3;
  animation: ${BoltAnimation} 2s linear infinite;

  &:nth-child(2) {
    width: 0.5em;
    height: 0.25em;
    margin: -1.75em 0 0 -1.875em;
    transform: translate(2.5em, 2.25em);
    opacity: 0.2;
    animation: ${BoltAnimation} 1.5s linear infinite;
  }
  &:before,
  &:after {
    content: '';
    position: absolute;
    z-index: 2;
    top: 50%;
    left: 50%;
    margin: -1.625em 0 0 -1.0125em;
    border-top: 1.25em solid transparent;
    border-right: 0.75em solid;
    border-bottom: 0.75em solid;
    border-left: 0.5em solid transparent;
    transform: skewX(-10deg);
  }
  &:after {
    margin: -0.25em 0 0 -0.25em;
    border-top: 0.75em solid;
    border-right: 0.5em solid transparent;
    border-bottom: 1.25em solid transparent;
    border-left: 0.75em solid;
    transform: skewX(-10deg);
  }
  &:nth-child(2):before {
    margin: -0.75em 0 0 -0.5em;
    border-top: 0.625em solid transparent;
    border-right: 0.375em solid;
    border-bottom: 0.375em solid;
    border-left: 0.25em solid transparent;
  }
  &:nth-child(2):after {
    margin: -0.125em 0 0 -0.125em;
    border-top: 0.375em solid;
    border-right: 0.25em solid transparent;
    border-bottom: 0.625em solid transparent;
    border-left: 0.375em solid;
  }
`;

Bolt.propTypes = {
  theme: ThemePropType,
};

Bolt.defaultProps = {
  theme: defaultTheme,
};

Bolt.displayName = 'Bolt';

export default Bolt;
