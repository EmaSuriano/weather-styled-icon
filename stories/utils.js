import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { WeatherThemeProvider } from '../src';
import { ThemePropType } from '../src/constants';

export const CenterContainer = styled.div`
  ${props =>
    props.theme.backgroundColor &&
    `background-color: ${props.theme.backgroundColor}`};
  text-align: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BackgroundContainer = styled(CenterContainer)`
  background-color: ${props => props.color};
`;

export const ThemeContainer = ({ theme, children }) => (
  <WeatherThemeProvider theme={theme}>
    <BackgroundContainer>{children}</BackgroundContainer>
  </WeatherThemeProvider>
);

ThemeContainer.propTypes = {
  theme: ThemePropType.isRequired,
  children: PropTypes.node,
};

ThemeContainer.defaultProps = {
  children: PropTypes.node.isRequired,
};
