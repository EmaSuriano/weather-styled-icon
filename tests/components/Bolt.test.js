import React from 'react';
import Bolt from '../../src/components/Bolt';
import { renderWithTheme } from '../utils';

describe('<Bolt />', () => {
  it('should should render div with theme', () => {
    const tree = renderWithTheme(<Bolt />);
    expect(tree).toMatchSnapshot();
  });

  xit('BoltAnimation should return keyframes with theme', () => {
    // TODO: Find a proper way to test Animations!
  });
});
