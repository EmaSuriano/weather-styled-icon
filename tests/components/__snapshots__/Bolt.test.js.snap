// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bolt /> should should render div with theme 1`] = `
.c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -0.25em 0 0 -0.125em;
  color: grey;
  opacity: 0.3;
  -webkit-animation: iiNwHN 2s linear infinite;
  animation: iiNwHN 2s linear infinite;
}

.c0:nth-child(2) {
  width: 0.5em;
  height: 0.25em;
  margin: -1.75em 0 0 -1.875em;
  -webkit-transform: translate(2.5em,2.25em);
  -ms-transform: translate(2.5em,2.25em);
  transform: translate(2.5em,2.25em);
  opacity: 0.2;
  -webkit-animation: iiNwHN 1.5s linear infinite;
  animation: iiNwHN 1.5s linear infinite;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  z-index: 2;
  top: 50%;
  left: 50%;
  margin: -1.625em 0 0 -1.0125em;
  border-top: 1.25em solid transparent;
  border-right: 0.75em solid;
  border-bottom: 0.75em solid;
  border-left: 0.5em solid transparent;
  -webkit-transform: skewX(-10deg);
  -ms-transform: skewX(-10deg);
  transform: skewX(-10deg);
}

.c0:after {
  margin: -0.25em 0 0 -0.25em;
  border-top: 0.75em solid;
  border-right: 0.5em solid transparent;
  border-bottom: 1.25em solid transparent;
  border-left: 0.75em solid;
  -webkit-transform: skewX(-10deg);
  -ms-transform: skewX(-10deg);
  transform: skewX(-10deg);
}

.c0:nth-child(2):before {
  margin: -0.75em 0 0 -0.5em;
  border-top: 0.625em solid transparent;
  border-right: 0.375em solid;
  border-bottom: 0.375em solid;
  border-left: 0.25em solid transparent;
}

.c0:nth-child(2):after {
  margin: -0.125em 0 0 -0.125em;
  border-top: 0.375em solid;
  border-right: 0.25em solid transparent;
  border-bottom: 0.625em solid transparent;
  border-left: 0.375em solid;
}

<div
  className="c0"
  data-e2e="bolt"
/>
`;
